eZ/Publish/Core/Search/Common/IndexerDataProvider.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Search\Common;
+
+use eZ\Publish\SPI\Search\IndexerDataProvider as SearchIndexerDataProvider;
+use eZ\Publish\SPI\Persistence\Handler;
+use eZ\Publish\SPI\Persistence\Content\ContentInfo;
+use eZ\Publish\Core\Persistence\Database\DatabaseHandler;
+use PDO;
+
+class IndexerDataProvider implements SearchIndexerDataProvider
```

Comment:
Our Search Engine Handler in this case is fully responsible for creating search engine index. Instead of adding to it a responsibility of fetching data, I think this data provider - hiding from Handler implementation details  - is better.

We could also keep fetching data in the command, but then we would have to do it separately for content and locations. Sth like this is done in #1738 so I don't think this PR would be an alternative if done like that.